src/objects/Player.gd

```diff
@@ -87,12 +103,18 @@ remote func set_network_transform(new_translation: Vector3, new_rotation: Vector
 
 
 func on_raycast_hit():
-	rpc("ive_been_hit")
-	ive_been_hit()
+	if is_vulnerable:
+		rpc("ive_been_hit")
+		ive_been_hit()
 
 
 remote func ive_been_hit():
 	$Blood.emitting = true
+	translation = get_tree().get_root().get_node("Game/Level/PlayerSpawnPoint").translation
```

Comment:
I wonder if we want to make the player disappear completely instead of be reset here. That's not a problem for this PR, just something we should think about later

Reply:
We could always do both?